server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -44,16 +45,21 @@
      * For convenience, the set of copied fields includes the field itself.
      */
     private final Map<String, Set<String>> fieldToCopiedFields = new HashMap<>();
-    private final String type;
     private final DynamicKeyFieldTypeLookup dynamicKeyLookup;
 
+    /**
+     * A field type representing the document type. This will be null for 8.0 indices,
+     * which don't have a type and do not support using the _type field in searches.
+     */
+    @Nullable private final TypeFieldType typeFieldType;
```

Comment:
@romseygeek I was wondering why we still have this logic on 'master' to handle searches on `_type`? Should it instead be removed completely?

Reply:
+1 to removing it entirely in master.  I think that was actually my original intention and I just never got round to it after I made the latest round of changes to TypeFieldType.

Reply:
Great, then I'll close this PR and open a new one to remove `TypeFieldType` entirely.

For context, what is the plan for handling 7.x API compatibility, where I assume `_type` will be allowed in searches? (I think I've asked you this 3+ times, I just always seem to forget...)

Reply:
I think it has changed a few times, @pgomulka will have a better idea of what the current plan is.

Reply:
@jtibshirani We plan to support `_type` fields when a request was made with compatible API. if it was on a request, or part of a path it would be parsed but the value ignored. If there will be a need to return a `_type` field on a response, we will return `_doc`.
It is great you are asking, I am planning to create few example Compatible API for types removal and ask for feedback (I have few stale PRs with examples, can share offline if you like)

Reply:
Thanks @pgomulka -- I will ping you offline to understand the plan. My question is if I can remove support for `_type` in searches entirely from master, or if I need to leave it in to coordinate with your plans.

Reply:
We discussed offline and agreed that the compatibility layer should continue to support `_type` within searches. However @pgomulka was fine with us removing this for now, and then later figuring out the best strategy for retaining compatibility (which may involve restoring some pieces).